modules/gapi/test/gapi_fluid_test.cpp

```diff
@@ -767,4 +767,28 @@ TEST(Fluid, UnusedNodeTest) {
         cv::compile_args(cv::gapi::core::fluid::kernels())));
 }
 
+TEST(Fluid, UnusedNodeReshapeTest) {
+    const auto test_size = cv::Size(8, 8);
+    const auto get_compile_args =
+        [] () { return cv::compile_args(cv::gapi::core::fluid::kernels()); };
+
+    cv::GMat in;
+    cv::GMat a, b, c, d;
+    std::tie(a, b, c, d) = cv::gapi::split4(in);
+    cv::GMat out = cv::gapi::resize(cv::gapi::merge3(a, b, c), test_size, 0.0, 0.0,
+        cv::INTER_LINEAR);
+    cv::GComputation comp(cv::GIn(in), cv::GOut(out));
+
+    cv::Mat in_mat(test_size, CV_8UC4);
+    cv::Mat out_mat(test_size, CV_8UC3);
+
+    cv::GCompiled compiled;
+    ASSERT_NO_THROW(compiled = comp.compile(descr_of(in_mat), get_compile_args()));
+
+    in_mat = cv::Mat(test_size * 2, CV_8UC4);
+    ASSERT_TRUE(compiled.canReshape());
+    ASSERT_NO_THROW(compiled.reshape(descr_of(gin(in_mat)), get_compile_args()));
+    ASSERT_NO_THROW(compiled(in_mat, out_mat));
```

Comment:
Probably .apply() already does that

Reply:
But it is still worth testing this way

Reply:
this is to ensure that after reshape the state of the execution is valid: in theory, we could've succeeded at compile/reshape stage but then fail at run, so decided to explicitly check execution as well